apps/tx-builder/src/typings/models.ts

```diff
@@ -1,13 +1,15 @@
-import { ContractMethod } from '../hooks/useServices/interfaceRepository';
+import { ContractMethod, ContractInterface } from '../hooks/useServices/interfaceRepository';
 
 export interface ProposedTransaction {
-  id: number | string;
+  id: number;
+  contractInterface: ContractInterface | null;
   description: {
     to: string;
     value: string;
-    hexEncodedData: string;
-    contractMethod: ContractMethod | undefined;
+    customTransactionData: string;
```

Comment:
Export file doesn't have to follow this interface 1:1

Reply:
Yes, I agree we don't need to follow exactly the same interface, just wondering if it would be a good improvement here too.

What you will agree is that the property should be at least optional in case you are using `contractMethod` and `contractInputsValues`

Reply:
good catch, I updated it